images/jupyter-hub/jupyterhub_config.py

```diff
@@ -166,6 +166,18 @@ def per_user_profiles(spawner):
 
 c.JupyterHub.authenticator_class = OrbitWorkbenchAuthenticator
 c.Authenticator.auto_login = True
+# Try to validate session / refresh token every 10 minutes
+c.Authenticator.auth_refresh_age = 600
+
+# Needed to save refresh token
+c.Authenticator.enable_auth_state = True
+
+if 'JUPYTERHUB_CRYPT_KEY' not in os.environ:
```

Comment:
who will export this env variable ?  do we need to add this to the script that starts jupyterhub on the jupyterhub image?  I don't see this change in this PR here.

Reply:
That's a great callout.  It wasn't obvious to me the best place to put it.  

One solution could create and persisting the encryption key in either Secrets Manager or Parameter Store.  Might be overkill if there will only be one JupyterHub instance at a time per team.  Open to suggestions.

As far as adding it to the image, it's a bit verbose to add an environment variable that's from a command:  https://stackoverflow.com/questions/34911622/dockerfile-set-env-to-result-of-command
like `openssl rand`

JupyterHub needs the key to encrypt the user auth store, since we'll be saving the jwt token per user.  

https://jupyterhub.readthedocs.io/en/stable/reference/authenticators.html#authentication-state

Reply:
Perhaps change Jupyterhub\DockerFile ,  add a script to run :
from:
   CMD ["jupyterhub", "-f", "/etc/jupyterhub/jupyterhub_config.py"]
 to:
   CMD ["./runJupyterHub.sh"]
 
and in runJupyterHub.sh:

export JUPYTERHUB_CRYPT_KEY=$(openssl rand -hex 32)
jupyterhub -f  /etc/jupyterhub/jupyterhub_config.py

(don't forget to add to Docker ADD command to add your new script and then we can rebuild using:
orbit build image -e dev-env -d ../images/jupyter-hub  -n jupyter-hub --debug

Reply:
Just added that to the image